apps/print/backend/src/app.ts

```diff
@@ -174,31 +179,73 @@ export function buildApi(ctx: AppContext) {
       return store.getBallots();
     },
 
-    async printBallot(input: { ballotPrintId: string }) {
+    getBallotPrintCounts({
+      precinctId,
+    }: {
+      precinctId?: PrecinctId;
+    }): BallotPrintCount[] {
+      return store.getBallotPrintCounts({ precinctId });
+    },
+
+    async printBallot(input: {
+      precinctId: Id;
+      splitId?: Id;
+      partyId?: Id;
+      languageCode: LanguageCode;
+      ballotType: BallotType;
+      copies: number;
+    }) {
+      const { electionDefinition } = assertDefined(store.getElectionRecord());
       const printerStatus = await printer.status();
       await logger.logAsCurrentRole(LogEventId.PrinterPrintRequest, {
-        message: `Attempting to print ballot with internal id ${input.ballotPrintId}`,
+        message: `Attempting to print ballot with ${input.copies} copies`,
+        ballotProps: JSON.stringify({
+          precinctId: input.precinctId,
+          splitId: input.splitId,
+          partyId: input.partyId,
+          languageCode: input.languageCode,
+          ballotType: input.ballotType,
+        }),
         printConnected: printerStatus.connected,
       });
-      const ballot = store.getBallot(input.ballotPrintId);
-      if (!ballot || !ballot.encodedBallot) {
-        await logger.logAsCurrentRole(LogEventId.PrinterPrintRequest, {
-          message: `No ballot found with id ${input.ballotPrintId}`,
-          disposition: 'failure',
-        });
-        return;
-      }
 
-      await printer.print({
-        data: Buffer.from(ballot.encodedBallot, 'base64'),
+      const ballotStyleId = findBallotStyleId(electionDefinition.election, {
+        precinctId: input.precinctId,
+        splitId: input.splitId,
+        languageCode: input.languageCode,
+        partyId: input.partyId,
+      });
+
+      // TODO(nikhil): Actually print ballot, increment print count within transaction
+      store.incrementBallotPrintCount({
+        precinctId: input.precinctId,
+        ballotStyleId,
+        ballotType: input.ballotType,
+        count: input.copies,
       });
+
+      // const ballot = store.getBallot(input.ballotPrintId);
+      // if (!ballot || !ballot.encodedBallot) {
+      //   await logger.logAsCurrentRole(LogEventId.PrinterPrintRequest, {
+      //     message: `No ballot found with id ${input.ballotPrintId}`,
+      //     disposition: 'failure',
+      //   });
+      //   return;
+      // }
+
+      // await printer.print({
+      //   data: Buffer.from(ballot.encodedBallot, 'base64'),
+      // });
```

Comment:
Are you keeping these commented out for the above TODO?

Reply:
Yeah, I was thinking it might be helpful for you if/when you are needing to test printing ballots. That being said what I left previously wasn't super helpful, but I added one more [commit](https://github.com/votingworks/vxsuite/pull/7538/commits/59529a6a786c4123eb3847129c45ae08156dba1a) that makes it so the commented out logic will work when uncommented